### PR TITLE
Add setLabelFromName

### LINKED ISSUE #77: Implement setLabelFromName & extractLabelFromName() functions
I'd like to be able to set a new label from series names so I can use it in the query (for example with ```groupBy```)

Example :
```
select(my.awesome.series.name.*.count)
  .where(...)
  .setLabelFromName("verb")
  .groupBy(verb, mean)

select(my.awesome.series.name.*.count)
  .where(...)
  .extractLabelFromName("verb","~MyRegex")
  .groupBy(verb, mean)
```

### PR DESCRIPTION
**Pull request by GillesBIANNIC**
fix #77

### CODE REVIEW

**Comment by aurrelhebert** [comment]
Could you add some documentation about this function here: https://github.com/ovh/tsl/blob/master/spec/doc.md#series-meta-operator

**Review by aurrelhebert — APPROVED** [review]

File: tsl/generateWarpScript.go [hunk 1]
```diff
@@ -1518,6 +1527,27 @@ func (protoParser *ProtoParser) renameTemplate(framework FrameworkStatement) (st
...
+	}
+
+	label := protoParser.getLit(params[0])
+	match := ""
+	if len(params) == 2 {
+		regex := protoParser.getLit(params[1])
+		match = regex + " MATCH DUP SIZE 0 > <% '' 0 SET %> IFT '' JOIN "
+	}
+
+	value = "<% DROP DUP LABELS 'labels' STORE DUP NAME 'name' STORE 'gts' STORE $labels $name " + match + label + " PUT 'labels' STORE $gts $labels RELABEL %> LMAP"
```
Comment (aurrelhebert) [inline tsl/generateWarpScript.go]: I thing here you can remove the LABELS GET call

Comment (aurrelhebert) [reply tsl/generateWarpScript.go]: And remove some of the WarpScript stored variables

Comment (GillesBIANNIC) [inline tsl/generateWarpScript.go]: I'll try to not use "STORE" and correctly use the stack instead

**Review by aurrelhebert — APPROVED** [review]

**Comment by aurrelhebert** [comment]
Thanks for your contribution :)

**Pull request merged**